### DISCUSSION TITLE
Event Addition: LoginDisconnectEvent

### DISCUSSION THREAD

**Discussion by AV3RG** [PaperMC/Paper#10510] [Ideas]
### Is your feature request related to a problem?

There is currently only one event that handles when player quits from the server, the `PlayerQuitEvent`. Unfortunately that event only works for players that were able to successfully connect to the server. Currently there is no event that is letting users handle disconnects from the server during the login phase.  

### Describe the solution you'd like.

We can add an event like `LoginDisconnectEvent` that lets plugin access the connection details, the disconnect reason and such. This can be a non-cancellable event as plugins should not be able to let them connect anyway.

### Describe alternatives you've considered.

We can do it via the `PreLogin` events but they don't let you cover disconnects due to things like `velocity.secret` verification, or player connecting without velocity.

**Comment by kennytv** [comment]
There's https://jd.papermc.io/paper/1.20/com/destroystokyo/paper/event/player/PlayerConnectionCloseEvent.html

**Comment by AV3RG** [comment]
This event is guaranteed to never fire unless [AsyncPlayerPreLoginEvent](https://jd.papermc.io/paper/1.20/org/bukkit/event/player/AsyncPlayerPreLoginEvent.html) has been fired beforehand. Are there any plans to implement it for events before that? Also attach a reason for connection closing with it
